docs/design/processing_trace_data/records_service.md

```diff
@@ -0,0 +1,214 @@
+
+## Frequency
+
+Frequency is defined as the number of events that occur in one second.
+
+### Example
+
+Input
+
+| Start timestamp |
+| --------------- |
+| 0.0             |
+| 0.1             |
+| 0.5             |
+| 1.2             |
+| 1.3             |
+| 2.3             |
+| ...             |
+
+Output
+
+| Timestamp | Period |
+| --------- | ------ |
+| 0.0       | 3.0    |
+| 1.0       | 2.0    |
+| ...       | ...    |
+
+## Latency
+
+Latency is defined as the time difference between the two columns of interest.
+
+$$
+latency_n = t^{end}_{n} - t^{start}_{n}
+$$
+
+### Example
+
+Input
+
+| Start timestamp | End timestamp |
+| --------------- | ------------- |
+| 0.0             | 0.1           |
+| 1.0             | 1.1           |
+| 2.0             | NaN           |
+| 3.0             | 3.1           |
+| ...             | ...           |
+
+Output
+
+| Start timestamp | Latency |
+| --------------- | ------- |
+| 0.0             | 0.1     |
+| 1.0             | 0.1     |
+| 3.0             | 0.1     |
+| ...             | ...     |
+
+## Response Time
+
+Response time is the amount of time it takes for a system to respond to an input.
+
+As shown above, latency can be calculated if a table can be constructed.
+However, the calculated latency is not suitable for evaluating response time.
+For example, if a sensor is driven at 10 Hz, a latency of up to 100 ms should be considered.
+
+CARET defines latency as the best-case response time and worst-case response time as the value that includes the delay due to cycles, as a more comprehensive approach.
+
+### Example
+
+Input
+
+| Start timestamp | End timestamp |
+| --------------- | ------------- |
+| 0.0             | 0.1           |
+| 1.0             | 1.1           |
+| 2.0             | NaN           |
+| 3.0             | 3.2           |
+| 4.0             | 4.3           |
+| ...             | ...           |
+
+Intermediate  
+Create intermediate data for when the interval in Start timestamp [0.0, 4.0] maps to End timestamp.
+
+| Start timestamp | End timestamp |
+| --------------- | ------------- |
+| [0.0, 1.0)      | 1.1           |
+| [1.0, 3.0)      | 3.2           |
+| [3.0, 4.0)      | 4.3           |
+
+Output
+
+| Start timestamp | Best-case response time | Worst-case response time |
+| --------------- | ----------------------- | ------------------------ |
+| 1.0             | 0.1 (1.1 - 1.0)         | 1.1 (1.1 - 0.0)          |
+| 3.0             | 0.2 (3.2 - 3.0)         | 2.3 (3.2 - 1.0)          |
+| ...             | ...                     | ...                      |
+
+Note that Best-case response time has the same definition as Latency.
+Worst-case response time also counts as response time in the case of a drop.
+
+### Visualize response time
+
+![Latency vs Response time](../../imgs/latency_vs_response_time.drawio.png)
+
+Cases like the one shown above include
```

Comment:
Of the pseudo two message flow diagram as shown above, the upper figure shows latency and the lower explains response time. 
The upper figure includes cumbersome cases as follows;